src/diff-snapshot.js

```diff
@@ -148,9 +148,8 @@ const alignImagesToSameSize = (firstImage, secondImage) => {
 
 const isFailure = ({ pass, updateSnapshot }) => !pass && !updateSnapshot;
 
-const shouldUpdate = ({ pass, updateSnapshot, updatePassedSnapshot }) => (
-  (!pass && updateSnapshot) || (pass && updatePassedSnapshot)
-);
+const shouldUpdate = ({ pass, updateSnapshot, updatePassedSnapshot }) =>
```

Comment:
This change should have required a unit test change, Please add unit tests that would fail if this code was reverted